src/main/resources/static/js/MainWindow.js

```diff
@@ -675,448 +761,479 @@ var initial = 1000;
+								.find("tbody tr")
+								.each(
+										function() {
+											currentInput = $(this).find(
+													"td:eq(1) input").val();
+											if (currentInput === null
+													|| currentInput === "") {
+												$
+														.notify(
+																{
+																	message : "Please specify the column where the observations of each datastream can be found (no empty field allowed) before importing data."
+																},
+																{
+																	type : "info",
+																	placement : {
+																		from : "top",
+																		align : "left"
+																	},
+																	animate : {
+																		enter : "animated fadeInDown",
+																		exit : "animated fadeOutUp"
+																	}
+																});
+												stop = true;
+												return false;
+											}
+
+											parsed = parseInt(currentInput, 10);
+
+											if (!(currentInput === parsed)
+													|| parsed < 0) {
+												$
+														.notify(
+																{
+																	message : "Please specify the column where the observations of each datastream can be found (must be a non-negative number) before importing data."
+																},
+																{
+																	type : "info",
+																	placement : {
+																		from : "top",
+																		align : "left"
+																	},
+																	animate : {
+																		enter : "animated fadeInDown",
+																		exit : "animated fadeOutUp"
+																	}
+																});
+												stop = true;
+												return false;
+											}
+											obs.push(parsed);
+										});
+						if (stop === true) {
+							return false;
+						}
+
+						obj["observations"] = obs;
+						obj["multiStream"] = (obs.length > 1);
+						streams.push(obj);
+					});
+	if (stop === true) {
+		return false;
+	}
+
+	var filetype = null;
+	if ($("input[name=source]:eq(0)").is(":checked")) {
+		filetype = $("#file").val();
+	} else if ($("input[name=source]:eq(1)").is(":checked")) {
+		filetype = $("#sourceinput").val();
+	}
+	if (!(filetype) || filetype === "") {
+		addToLog("No File specified.");
+	} else {
+		filetype = filetype.split(".").pop().toUpperCase();
+		if (filetype === "XLSX" || filetype === "XLS") {
+			filetype = "EXCEL";
+		} else if (filetype !== "CSV") {
+			addToLog("Unknown file type.");
+			return false;
+		}
+	}
+
+	var map = mappingData;
+
+	var formData = {
+		name : cfgName,
+		delimiter : currentDelimiter,
+		numberOfHeaderlines : parseInt(currentHeaderLines, 10),
+		timezone : $("#selecttime option:selected").attr("data-value"),
+		dateTime : date,
+		streamData : streams,
+		mapOfMagicNumbers : map,
+		dataType : filetype,
+		frostURL : url
+	};
+
+	var jsoncfg = JSON.stringify(formData, null, 4);
+	var mydata = {
+		config : jsoncfg,
+		filename : getCurrentFileName()
+	};
+	if ($("#selecttime option:selected").attr("data-value") == null) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FraunhoferIOSB/ChillImport/pullRequest?prid=4781409)